### ISSUE TITLE
Extract data adapter to proper repository and package

### ISSUE THREAD

**Issue by pjanuario** [rendrjs/rendr#457]
I think since the adapter could be a separated module, this would allow people to extend with new adapters and improve support on the existent one.

I implemented a data adapter that extends the RestAdapter and support query string authentication, but it's my intention to support (at least) basic auth and header authentication in a near future.

https://www.npmjs.com/package/rendr-auth-rest-adapter

To implement this I had to depend on rendr package, that is undesirable.

**Comment by saponifi3d** [comment]
I completely agree.  I'm working on trying to modularize all of Rendr, ideally the main repo will be bits and bobs of glue code to tie a couple default packages together.

**Comment by pjanuario** [comment]
+1...

@saponifi3d I don't know if you agree but I think a RestAdapter repository could have several implementations of rest adapter. I would expose the default implementation (the one that rendr have today) and people would be able to require the other implementations and use them. 

This would allows us to have a repo for rest adapter that have several implementations and get several contributions for different proposes. I have seen people making adapters for basic auth, token authentication on query string and/or header (me), cookie forwarding. 

I would be glad to make a PR with token auth code that I already have on my repo and later with header auth.
